boltdb.go

```diff
@@ -0,0 +1,248 @@
+/**
+* Copyright 2018 Comcast Cable Communications Management, LLC
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	bolt "github.com/coreos/bbolt"
+	"github.com/go-kit/kit/log/level"
+)
+
+// BoltDbCache describes a BoltDb Cache
+type BoltDbCache struct {
+	T      *TricksterHandler
+	Config BoltDbCacheConfig
+	dbh    *bolt.DB
+}
+
+// Connect instantiates the BoltDbCache mutex map and starts the Expired Entry Reaper goroutine
+func (c *BoltDbCache) Connect() error {
+
+	fullPath := path.Join(c.Config.CachePath, c.Config.Filename)
+
+	level.Info(c.T.Logger).Log("event", "boltdb cache setup", "cachePath", fullPath)
+
+	err := mustMakeDirectory(c.Config.CachePath)
```

Comment:
I'd like to avoid having Trickster create the database directory, this is the responsibility of the config management.

Reply:
I think that's fine, Prometheus does it too, right?

Reply:
Yea, it does, but Prometheus database is a lot more complicated than a single boltdb file. :-)